### PR TITLE
FOUNDATIONS: IServiceProvider part in InjectionDependencyService

### LINKED ISSUE #18: FOUNDATIONS: IServiceProvider part in `InjectionDependencyService`

### PR DESCRIPTION
**Pull request by LBoullosa**
closes #18

### CODE REVIEW

**Comment by glhays** [comment]
Will take a look at first chance. Your up late.


________________________________

File: STX.SPAL.Core.Tests.Unit/Services/Foundations/DependenciesInjections/DependencyInjectionServiceTests.Validations.BuildServiceProvider.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            this.dependencyInjectionBroker
+                .Verify(broker =>
+                    broker.BuildServiceProvider(
+                        It.IsAny<IServiceCollection>()),
+                Times.Never);
+
+            this.dependencyInjectionBroker.VerifyNoOtherCalls();
+        }
+
+
```
Comment (glhays) [inline STX.SPAL.Core.Tests.Unit/Services/Foundations/DependenciesInjections/DependencyInjectionServiceTests.Validations.BuildServiceProvider.cs]: Remove extra lines here.

File: STX.SPAL.Core.Tests.Unit/Services/Foundations/DependenciesInjections/DependencyInjectionServiceTests.cs [hunk 2]
```diff
@@ -371,5 +523,29 @@ public static TheoryData<Exception> RegisterServiceDescriptorServiceExceptions()
...
+            return new TheoryData<DependencyInjection, Xeption>
+            {
+                {
+                    null,
+                    CreateInvalidDependencyInjectionParameterException(
+                        new Dictionary<string, string>
+                        {
+                            {nameof(DependencyInjection), "object is required" }
+                        })
+
```
Comment (glhays) [inline STX.SPAL.Core.Tests.Unit/Services/Foundations/DependenciesInjections/DependencyInjectionServiceTests.cs]: Extra line

**Comment by glhays** [comment]
@LBoullosa just a heads up on current and upcoming Standard upgrades that have and will be integrated in our work. You know the ValueTask Async v2.10.2, following in the works is v2.11.0 Api Controller Testing (not applicable here) and in the works also is something Hassan is calling localize->localize is v2.11.1
Just letting you know because you started back working on this. Maybe if applicable you want to integrate v2.10.2 and v2.11.1. Now or later???

File: STX.SPAL.Core.Tests.Unit/Services/Foundations/DependenciesInjections/DependencyInjectionServiceTests.Validations.BuildServiceProvider.cs (hunk 1 above)
Comment (LBoullosa) [reply STX.SPAL.Core.Tests.Unit/Services/Foundations/DependenciesInjections/DependencyInjectionServiceTests.Validations.BuildServiceProvider.cs]: Done.

File: STX.SPAL.Core.Tests.Unit/Services/Foundations/DependenciesInjections/DependencyInjectionServiceTests.cs (hunk 2 above)
Comment (LBoullosa) [reply STX.SPAL.Core.Tests.Unit/Services/Foundations/DependenciesInjections/DependencyInjectionServiceTests.cs]: Done.

**Comment by LBoullosa** [comment]
> @LBoullosa just a heads up on current and upcoming Standard upgrades that have and will be integrated in our work. You know the ValueTask Async v2.10.2, following in the works is v2.11.0 Api Controller Testing (not applicable here) and in the works also is something Hassan is calling localize->localize is v2.11.1 Just letting you know because you started back working on this. Maybe if applicable you want to integrate v2.10.2 and v2.11.1. Now or later???

Correct, I thought about it. I will go into it when this foundation was completed.

**Review by glhays — APPROVED** [review]

**Pull request merged**